src/adapter/templates/getStringyProps.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { remoteFunction } from '.';
+
+/**
+ * Gets a mapping of property names with a custom `.toString()` method
+ * to their string representations.
+ */
+export const getStringyProps = remoteFunction(function (this: unknown, maxLength: number) {
+  const out: Record<string, string> = {};
+  if (typeof this !== 'object' || !this) {
+    return out;
+  }
+
+  for (const [key, value] of Object.entries(this)) {
+    if (typeof value === 'object' && value && !String(this.toString).includes('[native code]')) {
```

Comment:
This detection is a bit ugly. Not sure if there's a more clever way to do this. Simply `this.toString === Object.prototype.toString` isn't enough because that also incorrectly stringifies boxed types, errors, and arrays.